### PR TITLE
feat(telecom): add softphone configuration

### PR DESCRIPTION
**Pull request by antonymarion**
ref: MANAGER-13662

| Question         | Answer
| ---------------- | ---
| Branch?          | `master`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Feat#MANAGER-13662
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally (**on LABEU, should be tested on real API then**)
- [x] Ticket reference is mentioned in linked commits (internal only)
~~- [ ] Breaking change is mentioned in relevant commits~~

### CODE REVIEW

File: packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/softphone.controller.js [hunk 1]
```diff
@@ -16,17 +17,62 @@ export default class SoftphoneController {
...
     this.MOBILE_OS = MOBILE_OS;
     this.guide = {
-      title: 'mocked_title',
-      url: 'mocked_url',
-      external: 'mocked_external',
-      name: 'mocked_name',
+      title: 'telephony_line_softphone_guide_title',
+      url: GUIDE_URL,
+      external: GUIDE_URL,
+      name: 'telephony_line_softphone_configuration_name',
```
Comment (ghyenne) [inline packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/softphone.controller.js:31]:
```suggestion
      title: 'telephony_line_softphone_configuration_name',
      url: GUIDE_URL,
      external: true,
```
The translate of button "guide" can be put directly in html. External require a boolean

File: packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/softphone.html [hunk 2]
```diff
@@ -79,11 +79,11 @@ <h4 class="pt-3">
             <p
                 data-translate="telephony_line_softphone_configure_description"
             ></p>
-            <oui-guide-menu data-text="{{ ::$ctrl.guide.title }}">
+            <oui-guide-menu data-text="{{ ::$ctrl.guide.title | translate }}">
                 <oui-guide-menu-item
                     data-href="{{ ::$ctrl.guide.url }}"
                     data-external="$ctrl.guide.external"
-                    ><span data-ng-bind="::$ctrl.guide.name"></span>
+                    ><span data-ng-bind="::$ctrl.guide.name | translate"></span>
```
Comment (ghyenne) [inline packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/softphone.html:86]:
```suggestion
                    ><span data-translate="{{ $ctrl.guide.name }}"></span>
```

File: packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/softphone.html [hunk 3]
```diff
@@ -93,9 +93,65 @@ <h4 class="pt-3">
...
+                                    data-ng-click=":: $ctrl.cancelEdition(deviceId)"
+                                    class="oui-icon oui-icon-close text-secondary"
+                                    aria-hidden="true"
+                                ></span>
+                            </div>
+                        </div>
+                    </form>
+                </div>
+                <oui-tile-definition
+                    term="{{:: 'telephony_line_softphone_status' | translate}}"
```
Comment (ghyenne) [inline packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/softphone.html:140]:
```suggestion
                   data-term="{{:: 'telephony_line_softphone_status' | translate}}"
```

File: packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/softphone.routing.js [hunk 4]
```diff
@@ -45,6 +45,18 @@ export default /* @ngInject */ ($stateProvider) => {
...
+          softphoneDevices,
+          billingAccount,
+          serviceName,
+          $q,
+        ) =>
+          $q.all(
+            softphoneDevices.map((deviceId) =>
+              softphoneService.getDevice(billingAccount, serviceName, deviceId),
+            ),
+          ),
```
Comment (ghyenne) [inline packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/softphone.routing.js:59]: You have to do a iceberg call not one for id and second for expend.

File: packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/softphone.html [hunk 5]
```diff
@@ -79,11 +79,11 @@ <h4 class="pt-3">
             <p
                 data-translate="telephony_line_softphone_configure_description"
             ></p>
-            <oui-guide-menu data-text="{{ ::$ctrl.guide.title }}">
+            <oui-guide-menu data-text="{{ ::$ctrl.guide.title | translate }}">
```
Comment (ghyenne) [inline packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/softphone.html:82]:
```suggestion
            <oui-guide-menu data-text="{{ 'telephony_line_softphone_guide_title' | translate }}">
```

File: packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/softphone.html [hunk 6]
```diff
@@ -93,9 +93,65 @@ <h4 class="pt-3">
...
+                >
+                    <oui-tile-description>
+                        <div class="mt-1">
+                            <span
+                                data-ng-class="{
+                                     'bg-danger': !$ctrl.getDeviceStatusColorById(deviceId),
+                                     'bg-success': $ctrl.getDeviceStatusColorById(deviceId)
+                                     }"
+                                class="rounded p-1 text-white"
+                                data-translate="::$ctrl.getDeviceStatusById(deviceId)"
```
Comment (ghyenne) [inline packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/softphone.html:150]:
```suggestion
                                data-translate="{{ 'my_key_for_status_' + $ctrl.deviceId.status }}"
```

File: packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/softphone.html [hunk 7]
```diff
@@ -93,9 +93,65 @@ <h4 class="pt-3">
...
+                <oui-tile-definition
+                    term="{{:: 'telephony_line_softphone_status' | translate}}"
+                >
+                    <oui-tile-description>
+                        <div class="mt-1">
+                            <span
+                                data-ng-class="{
+                                     'bg-danger': !$ctrl.getDeviceStatusColorById(deviceId),
+                                     'bg-success': $ctrl.getDeviceStatusColorById(deviceId)
+                                     }"
```
Comment (ghyenne) [inline packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/softphone.html:148]:
```suggestion
                                data-ng-class="{
                                     'oui-badge_error': $ctrl.deviceId.status === $ctrl.MY_CONST_FOR_STATUS.unregistered,
                                     'oui-badge_success': $ctrl.deviceId.status === $ctrl.MY_CONST_FOR_STATUS.registered,
                                     }"
```

File: packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/softphone.html [hunk 8]
```diff
@@ -93,9 +93,65 @@ <h4 class="pt-3">
...
+                                class="rounded p-1 text-white"
```
Comment (ghyenne) [inline packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/softphone.html:149]:
```suggestion
                                class="oui-badge"
```

File: packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/softphone.html [hunk 9]
```diff
@@ -93,9 +93,65 @@ <h4 class="pt-3">
...
+                                    data-ng-click=":: $ctrl.saveSoftphoneName(deviceId)"
+                                    data-ng-disabled="!editableServiceNameForm.$valid"
+                                ></span>
+                                <span
+                                    data-ng-click=":: $ctrl.cancelEdition(deviceId)"
+                                    class="oui-icon oui-icon-close text-secondary"
+                                    aria-hidden="true"
+                                ></span>
+                            </div>
+                        </div>
```
Comment (ghyenne) [inline packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/softphone.html:136]: Maybe you can use oui-field ui-kit component

**Review by ghyenne — CHANGES_REQUESTED** [review]

File: packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/softphone.service.js [hunk 10]
```diff
@@ -8,6 +8,14 @@ export default class SofpthoneService {
...
   }
 
+  getDevice(billingAccount, serviceName, deviceId) {
+    return this.$http
+      .get(
+        `/telephony/${billingAccount}/line/${serviceName}/devices/${deviceId}`,
+      )
+      .then(({ data }) => data);
+  }
+
```
Comment (ghyenne) [inline packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/softphone.service.js:18]:
```suggestion
```

File: packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/softphone.controller.js [hunk 11]
```diff
@@ -16,17 +18,54 @@ export default class SoftphoneController {
...
+      .getDevicesInfos(this.billingAccount, this.serviceName)
+      .then((devices) => {
+        this.isLoading = false;
+        this.devices = devices;
+      });
+  }
+
+  razDeviceIdModelName(deviceId) {
+    this.model[deviceId] = { name: '' };
+  }
```
[the remark this answers is not in the document]
Comment (antonymarion) [reply packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/softphone.controller.js:49]: I believe I must keep this one

File: packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/softphone.controller.js [hunk 12]
```diff
@@ -16,17 +18,54 @@ export default class SoftphoneController {
...
+
+  cancelEdition(deviceId) {
+    this.showEditFormSoftphoneName[deviceId] = false;
+    this.razDeviceIdModelName(deviceId);
+  }
```
[the remark this answers is not in the document]
Comment (antonymarion) [reply packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/softphone.controller.js:54]: I believe I must keep this one

File: packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/softphone.controller.js [hunk 13]
```diff
@@ -16,17 +18,54 @@ export default class SoftphoneController {
...
+      this.serviceName,
+      this.model[deviceId].name,
+      deviceId,
+    );
+  }
+
+  toggleEditSoftphoneName(deviceId) {
+    this.showEditFormSoftphoneName[deviceId] = !this.showEditFormSoftphoneName[
+      deviceId
+    ];
```
[the remark this answers is not in the document]
Comment (antonymarion) [reply packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/softphone.controller.js:77]: image
image

I am keeping model.name for readability (ease comprehension of property modified on the model)

**Review by ghyenne — APPROVED** [review]

**Pull request merged**